stub-idp/src/main/java/uk/gov/ida/stub/idp/repositories/StubCountryRepository.java

```diff
@@ -1,23 +1,34 @@
 package uk.gov.ida.stub.idp.repositories;
 
+import uk.gov.ida.stub.idp.configuration.IdpStubsConfiguration;
+import uk.gov.ida.stub.idp.configuration.StubIdp;
+import uk.gov.ida.stub.idp.configuration.UserCredentials;
 import uk.gov.ida.stub.idp.domain.EidasScheme;
+import uk.gov.ida.stub.idp.domain.factories.IdpStubsConfigurationFactory;
 
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.ws.rs.core.UriBuilder;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 import static java.text.MessageFormat.format;
 
 public class StubCountryRepository {
 
+    public static final String STUB_COUNTRY_FRIENDLY_ID = EidasScheme.stub_country.getEidasSchemeName();
+
+    private final IdpStubsConfigurationFactory idpStubsConfigurationFactory;
     private final AllIdpsUserRepository allIdpsUserRepository;
     private final String stubCountryMetadataUrl;
-    public static final String STUB_COUNTRY_FRIENDLY_ID = EidasScheme.stub_country.getEidasSchemeName();
 
     @Inject
-    public StubCountryRepository(AllIdpsUserRepository allIdpsUserRepository, @Named("StubCountryMetadataUrl") String stubCountryMetadataUrl) {
+    public StubCountryRepository(AllIdpsUserRepository allIdpsUserRepository, @Named("StubCountryMetadataUrl") String stubCountryMetadataUrl, IdpStubsConfigurationFactory idpStubsConfigurationFactory) {
         this.allIdpsUserRepository = allIdpsUserRepository;
         this.stubCountryMetadataUrl = stubCountryMetadataUrl;
+        this.idpStubsConfigurationFactory = idpStubsConfigurationFactory;
         allIdpsUserRepository.createHardcodedTestUsersForCountries(STUB_COUNTRY_FRIENDLY_ID, STUB_COUNTRY_FRIENDLY_ID);
```

Comment:
This all looks a bit hard-coded to assuming there's one stub-country. Can we remove this assumption?